x-pack/plugins/aiops/public/ui_actions/create_pattern_analysis_action.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import type { PresentationContainer } from '@kbn/presentation-containers';
+import type { EmbeddableApiContext } from '@kbn/presentation-publishing';
+import type { UiActionsActionDefinition } from '@kbn/ui-actions-plugin/public';
+import { IncompatibleActionError } from '@kbn/ui-actions-plugin/public';
+import type { CoreStart } from '@kbn/core-lifecycle-browser';
+import { EMBEDDABLE_PATTERN_ANALYSIS_TYPE } from '@kbn/aiops-log-pattern-analysis/constants';
+import type { AiopsPluginStartDeps } from '../types';
+import type { PatternAnalysisActionContext } from './pattern_analysis_action_context';
+import type {
+  PatternAnalysisEmbeddableApi,
+  PatternAnalysisEmbeddableInitialState,
+} from '../embeddables/pattern_analysis/types';
+
+const parentApiIsCompatible = async (
+  parentApi: unknown
+): Promise<PresentationContainer | undefined> => {
+  const { apiIsPresentationContainer } = await import('@kbn/presentation-containers');
+  // we cannot have an async type check, so return the casted parentApi rather than a boolean
+  return apiIsPresentationContainer(parentApi) ? (parentApi as PresentationContainer) : undefined;
+};
+
+export function createAddPatternAnalysisEmbeddableAction(
+  coreStart: CoreStart,
+  pluginStart: AiopsPluginStartDeps
+): UiActionsActionDefinition<PatternAnalysisActionContext> {
+  return {
+    id: 'create-pattern-analysis-embeddable',
+    grouping: [
+      {
+        id: 'ml',
+        getDisplayName: () =>
+          i18n.translate('xpack.aiops.navMenu.mlAppNameText', {
+            defaultMessage: 'Machine Learning and Analytics',
+          }),
+        getIconType: () => 'machineLearningApp',
+      },
+    ],
+    getIconType: () => 'machineLearningApp',
```

Comment:
Please use the dedicated icon Joana created for this, `logPatternAnalysis`
```suggestion
    getIconType: () => 'logPatternAnalysis',
```

Reply:
Updated in 09fe9b8c102582d77cd7a522ea890869360b6dd2